include/dlaf/matrix/internal/tile_future_manager.h

```diff
@@ -17,15 +17,19 @@ namespace matrix {
 namespace internal {
 
 // Attach the promise to the tile included in old_future with a continuation and returns a new future to it.
-template <class ReturnTileType, class TileType>
-hpx::future<ReturnTileType> setPromiseTileFuture(hpx::future<TileType> old_future,
-                                                 hpx::lcos::local::promise<TileType> p) noexcept {
+template <class ReturnTileType>
+hpx::future<ReturnTileType> setPromiseTileFuture(
+    hpx::future<typename ReturnTileType::TileDataType> old_future,
+    hpx::lcos::local::promise<typename ReturnTileType::TileDataType> p) noexcept {
+  using TileDataType = typename ReturnTileType::TileDataType;
+  using NonConstTileType = typename ReturnTileType::TileType;
+
   DLAF_ASSERT_HEAVY(old_future.valid(), "");
-  return old_future.then(hpx::launch::sync, [p = std::move(p)](hpx::future<TileType>&& fut) mutable {
+  return old_future.then(hpx::launch::sync, [p = std::move(p)](hpx::future<TileDataType>&& fut) mutable {
     std::exception_ptr current_exception_ptr;
 
     try {
-      return ReturnTileType(std::move(fut.get().setPromise(std::move(p))));
+      return ReturnTileType(std::move(NonConstTileType(std::move(fut.get())).setPromise(std::move(p))));
```

Comment:
Are all the nested constructors and `move`s really necessary? The inner `move` around `fut.get()` should be unnecessary. See also other comment about the return type of `setPromise`.

Reply:
You are correct, one of the move is not necessary.